### DISCUSSION TITLE
Does WingetUI support updating global NPM packages?

### DISCUSSION THREAD

**Discussion by edumserrano** [Devolutions/UniGetUI#2152] [Q&A]
It would be nice if WingetUI would support updating global NPM packages. That doesn't seem to be the case or at least I'm not getting suggestions for some packages I know could be updated so either I have some other issue or this feature is not supported?

**Comment by marticliment** [comment]
They should be updated, please open a new issue about it

**Reply by edumserrano** [reply]
I noticed on the WingetUI Logs that it executes the following command: `npm outdated --global` which results in:

```text
▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
█▀▀▀▀▀▀▀▀▀ [19/04/2024 23:37:00] Npm ▀▀▀▀▀▀▀▀▀▀▀
█  Executable: C:\Windows\system32\windowspowershell\v1.0\powershell.exe
█  Arguments:  -NoProfile -ExecutionPolicy Bypass -Command npm outdated --global --parseable
[19/04/2024 23:37:00] Exit Code: 0
▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
```

If I run the same command on Windows Terminal I indeed get no results back:

![image](https://github.com/marticliment/WingetUI/assets/15857357/94e4e5c8-f746-4561-8461-fda0eb981b53)

I did a quick search on issues with `npm outdated` and it seems there's an open bug about it npm/cli#6098. As the users in this issue I am running on Windows and using NVM (not sure it matters but thought I should point this out). See [this comment](https://github.com/npm/cli/issues/6098#issuecomment-1413943569). It seems this issue with `npm outdated` is also affecting the [npm-gui](https://github.com/q-nick/npm-gui) app.

It seems this is NOT a bug with WingetUI but more that the way WingetUI checks for updates on global packages needs to change? For instance, if I use [npm-check-updates](https://www.npmjs.com/package/npm-check-updates) by running: `npx npm-check-updates -g` I get the expected results of global installed packages to update:

```text
[====================] 5/5 100%

 @angular/cli   17.3.2  →  17.3.5
 corepack       0.23.0  →  0.27.0
 npm            10.4.0  →  10.5.2
 vsts-npm-auth  0.42.1  →  0.43.0

ncu itself cannot upgrade global packages. Run the following to upgrade all global packages:

npm -g install @angular/cli@17.3.5 corepack@0.27.0 npm@10.5.2 vsts-npm-auth@0.43.0
```

What are your thoughts?

**Reply by marticliment** [reply]
I will investigate this
